### PR TITLE
Implement sorting of App admin object tables by column ascending/descending (#120)

### LINKED ISSUE #120: Allow Works to be sorted
In the overview of [Works](https://staging.thoth.openbookpublishers.com/admin/works) it would be useful if you could sort the different columns ascending and descending.

### PR DESCRIPTION
**Pull request by rhigman**
Fixes #120. Also includes adding "UpdatedAt" date column to each table. Builds on backend changes to implement date columns and sorting in graphql/database under #153, #156, #158.

Behaviour:
- Clicking on a column header which corresponds to an object field (e.g. `Title` in `Works` table) sorts the table by that column, ascending.
- Clicking on the same column header again reverses the sort order, e.g. clicking twice in a row sorts the table descending.
- Column headers which correspond to object relations rather than fields (e.g. `Contributors` in `Works` table) cannot be clicked/sorted.

An alternative implementation would have been to add up/down arrow icons to the column headers to represent clickable ascending/descending sort buttons. This is not a native Bulma table feature, but some implementation examples can be found at https://github.com/bokuweb/re-bulma-sortable-table and https://github.com/JDilleen/datatables-bulma if we want to go down this route.

Changes include:
- Moving `OrderBy` structs into individual API object models to allow their reuse in App, and implementing defaults
- Replacing existing API enum `FromStr` and `Display` boilerplate with `strum` auto-derives
- Converting `NaiveDateTime`s to `DateTime<Utc>`s for more human-readable JSON (Diesel appears to automatically convert `Timestamp`s into `Timestamptz`s so no data type changes were required in database)

### CODE REVIEW

File: thoth-app/src/component/mod.rs [hunk 1]
```diff
@@ -276,7 +300,25 @@ macro_rules! pagination_component {
                                             <tr>
                                                 {
                                                     for self.table_headers.iter().map(|h| {
-                                                        html! {<th>{h}</th>}
+                                                        {
+                                                            // If the header is a sortable field, make it clickable
+                                                            match <$order_field>::from_str(&h) {
+                                                                Ok(header) => {
+                                                                    html! {
+                                                                        <th
```
Comment (ja573) [inline thoth-app/src/component/mod.rs:308]: We should add the `is-clickable` class here to make it obvious to the user that clicking on the columns will do something

File: thoth-api/src/series/model.rs [hunk 2]
```diff
@@ -25,12 +27,20 @@ pub enum SeriesType {
...
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, EnumString, Display)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 pub enum SeriesField {
+    #[serde(rename = "SERIES_ID")]
+    #[strum(serialize = "ID")]
     SeriesID,
     SeriesType,
+    #[strum(serialize = "Series")]
     SeriesName,
+    #[serde(rename = "ISSNPRINT")]
```
Comment (ja573) [inline thoth-api/src/series/model.rs:39]: I hadn't realised these were the enum values we had for ISSN. It'd make more sense to have them as `ISSN_PRINT` and `ISSN_DIGITAL`

File: thoth-app/src/models/contributor/mod.rs [hunk 3]
```diff
@@ -55,11 +58,26 @@ impl Contributor {
...
+                <td>{&self.updated_at.format("%F %T")}</td>
             </tr>
         }
     }
 }
 
+impl Default for Contributor {
+    fn default() -> Contributor {
+        Contributor {
+            contributor_id: "".to_string(),
```
Comment (ja573) [inline thoth-app/src/models/contributor/mod.rs:70]: just fyi (no need to change anything), you can call `Default:::default()` on these fields individually, e.g. `contributor_id: Default::default()`, as strings and options do implement Default

File: thoth-api/src/series/model.rs (hunk 2 above)
Comment (rhigman) [reply thoth-api/src/series/model.rs]: Agreed - couldn't remember how to change this when I first encountered it, but I've sorted it now :)

**Review by ja573 — APPROVED** [review]

**Pull request merged**